cmd/aad-cli/cli/user_test.go

```diff
@@ -0,0 +1,200 @@
+package cli_test
+
+import (
+	"context"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/aad-auth/cmd/aad-cli/cli"
+	"github.com/ubuntu/aad-auth/internal/cache"
+	"github.com/ubuntu/aad-auth/internal/testutils"
+)
+
+func TestUserShellCompletion(t *testing.T) {
+	tests := map[string]struct {
+		args []string
+	}{
+		"get all users for get, short flag":    {args: []string{"user", "-n"}},
+		"get all users for get, long flag":     {args: []string{"user", "--name"}},
+		"get attributes for user":              {args: []string{"user"}},
+		"get attributes for overridden user":   {args: []string{"user", "--name", "myuser@domain.com"}},
+		"default completion for last argument": {args: []string{"user", "gecos"}},
+		"default completion, overridden user":  {args: []string{"user", "gecos", "--name", "myuser@domain.com"}},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			args := []string{cobra.ShellCompRequestCmd}
+			args = append(args, tc.args...)
+			args = append(args, "")
+
+			cacheDir := t.TempDir()
+			testutils.CopyDBAndFixPermissions(t, filepath.Join("testdata", "cachedb"), cacheDir)
+
+			uid, gid := testutils.GetCurrentUIDGID(t)
+
+			cache, err := cache.New(context.Background(), cache.WithCacheDir(cacheDir), cache.WithRootUID(uid), cache.WithRootGID(gid), cache.WithShadowGID(gid))
+			require.NoError(t, err, "Setup: failed to create cache")
+
+			c := cli.New(cli.WithCache(cache))
```

Comment:
Ah, I understand now why you are attaching cache to the app itself. Ack on it then. (still, my comments to remove option is valid). I kept my previous remark to see that I was first puzzled by it and that maybe a comment will be beneficial?